src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -2498,6 +2498,51 @@ class TypeSpecTest {
     }
   }
 
+  @Test fun classExtendsNoPrimaryConstructor() {
+    val typeSpec = TypeSpec.classBuilder("IoException")
+        .superclass(Exception::class)
+        .addFunction(FunSpec.constructorBuilder().build())
+        .build()
+
+    assertThat(toString(typeSpec)).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import java.lang.Exception
+      |
+      |class IoException : Exception {
+      |  constructor() {
+      |  }
+      |}
+      |""".trimMargin())
+  }
+
+  @Test fun classExtendsNoPrimaryOrSecondaryConstructor() {
+    val typeSpec = TypeSpec.classBuilder("IoException")
+        .superclass(Exception::class)
+        .build()
+
+    assertThat(toString(typeSpec)).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import java.lang.Exception
+      |
+      |class IoException : Exception()
+      |""".trimMargin())
+  }
+
+  @Test fun classExtendsNoPrimaryConstructorButSuperclassParams() {
+    try {
+      TypeSpec.classBuilder("IoException")
+          .superclass(Exception::class)
+          .addSuperclassConstructorParameter("%S", "hey")
+          .addFunction(FunSpec.constructorBuilder().build())
+          .build()
```

Comment:
Missing a fail

Reply:
Bah we need a lambda version of this assertion so I don't have to bother